201-vm-vsts-agent/azuredeploy.json

```diff
@@ -1,13 +1,33 @@
 {
     "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
     "contentVersion": "1.0.0.0",
-    "parameters": {        
+    "parameters": {
+        "modules": {
+            "type": "array",
+            "metadata": {
+                "description": "Enter the Module(s) to install including the Version"
+            },
+            "defaultValue": [
+                {"name": "AzureRM", "version": "5.6.0"},
+                {"name": "AzureAD", "version": "2.0.1.3"},
+                {"name": "Bitbucket.v2", "version": "1.1.2"},
+                {"name": "GetPassword", "version": "1.0.0.0"},
+                {"name": "posh-git", "version": "0.7.1"}
+            ]
+        },
         "publicIPDnsName": {
             "type": "string",
             "metadata": {
                 "description": "Enter a DNS name to resolve to the Public IP Address"
             }
         },
+        "scriptUrl": {
```

Comment:
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc) for how to handle artifacts